### PR TITLE
Assignment3 @hgill17 @JessicaH46

### PR DESCRIPTION
**Pull request by dcox2005**
API: https://wordleapiclassdavid.azurewebsites.net/swagger
Website: https://icy-pond-058a8a210.1.azurestaticapps.net/

### CODE REVIEW

**Comment by kelseymcmahon** [comment]
Everything looked good! I liked that their is an option to view all the players on the scoreboard. It looks like the top ten is not ordered by the best to worst players, this would be nice to add.

File: wordle-web/pages/leaderboard.vue [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  </v-container>
+</template>
+
+<script lang="ts">
+import { Component, Vue } from 'vue-property-decorator'
+
+@Component({})
+export default class leaderboard extends Vue {
+  players: any = []
+
```
Comment (mmwoodfo) [inline wordle-web/pages/leaderboard.vue:49]:
```suggestion
  created(){
    this.getAllPlayers();
  }
```
For a better user experience, I would recommend pre-loading some data onto the table when the page loads using the `created` method.

File: wordle-web/pages/leaderboard.vue [hunk 2]
```diff
@@ -0,0 +1,62 @@
+<template>
+  <v-container fluid fill-height justify-center>
+    <v-card>
+      <v-card-title class="display-3 justify-center">
+        Leader Board
+      </v-card-title>
+      <v-card-text>
+        <v-simple-table>
```
Comment (mmwoodfo) [inline wordle-web/pages/leaderboard.vue:8]: Good use of the vuetify simple table. :)

File: wordle-web/pages/leaderboard.vue [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+          <thead>
+            <tr>
+              <th>Name</th>
+              <th style="text-align: center"># Games</th>
+              <th style="text-align: center">Avg. Attempts</th>
+              <th style="text-align: center">Avg. Seconds</th>
```
Comment (mmwoodfo) [inline wordle-web/pages/leaderboard.vue:14]:
```suggestion
              <th class="text-center"># Games</th>
              <th class="text-center">Avg. Attempts</th>
              <th class="text-center">Avg. Seconds</th>
```

Avoid using inline styles & instead take advantage of either pre-built CSS classes from vuetify, or your own CSS classes.

File: wordle-web/pages/leaderboard.vue [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+          <tbody>
+            <tr v-for="(player, playerId) in players" :key="playerId">
+              <td>{{ player.name }}</td>
+              <td style="text-align: center">{{ player.gameCount }}</td>
+              <td style="text-align: center">
+                {{ player.averageAttempts.toFixed(2) }}
+              </td>
+              <td style="text-align: center">
+                {{ player.averageSecondsPerGame }}
+              </td>
```
Comment (mmwoodfo) [inline wordle-web/pages/leaderboard.vue:26]:
```suggestion
              <td class="text-center">{{ player.gameCount }}</td>
              <td class="text-center">
                {{ player.averageAttempts.toFixed(2) }}
              </td>
              <td class="text-center">
                {{ player.averageSecondsPerGame }}
              </td>
```

File: wordle-web/pages/game.vue [hunk 5]
```diff
@@ -35,20 +35,52 @@
             <span> Go Home </span>
           </v-tooltip>
         </v-col>
+        <v-col cols="5" class="d-flex flex-row-reverse">
```
Comment (mmwoodfo) [inline wordle-web/pages/game.vue:38]: You may consider making this into it's own component.

Comment (mmwoodfo) [reply wordle-web/pages/game.vue:38]: Additionally, if the username is 'Guest' the game should prompt for a name at the end. (dialog should pop up before the save occurs.)

File: .github/workflows/main_wordleapiclassdavid.yml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+on:
+  push:
+    branches:
+      - Assignment3
+    paths:
+      - 'Wordle.api/**'
+  workflow_dispatch:
+
+jobs:
+  build:
```
Comment (mmwoodfo) [inline .github/workflows/main_wordleapiclassdavid.yml:15]: Doesn't look like you're doing a `dotnet test` anywhere?

**Pull request closed without merging**